### ISSUE TITLE
Extend beyond mpris?

### ISSUE THREAD

**Issue by mirh** [FichteFoll/discordrp-mpris#12]
So, I don't know if this could even be in your interests..
Still I found your service is like the only one on the whole internet that "takes a standard agnostic format" and converts it to DRPC (rather than just directly hooking into programs and sending it directly to it). 
And I was wondering if your fine work couldn't be made a bit more general-purpose (or something at least enough for games I guess). 

Unfortunately there isn't really an xdg specification specific to this, still d-bus [is](https://telepathy.freedesktop.org/spec/Connection_Interface_Simple_Presence.html) [extensible](https://dbus.freedesktop.org/doc/dbus-api-design.html) [enough](https://techbase.kde.org/Development/Tutorials/D-Bus/Creating_Interfaces) that I'd hope it shouldn't be too hard to design a simple custom api.

**Comment by FichteFoll** [comment]
I'm not really sure what you're asking for. This tool was made specifically for the mpris interface and uses Discord's RPC protocol to provide the necessary information to the Discord client. The protocol is documented and has a reference implementation in C, as well as a couple implementations in different languages. At the time I worked on this, I didn't find an existing library to my liking (especially one supporting asyncio), so I reused my own implementation I made for https://github.com/Snazzah/SublimeDiscordRP. In the meantime, more libraries have been written and I have collected references to those over time that I will reevaluate when I reserve time to work on this again. I remember at least one or two seeming promising and allowing me to drop my custom and occasionally flaky implementation.

If you're looking for a generic dbus proxy interface to access Discord's rich presence protocol, I suggest using or writing a different application, as that would be entirely out of scope.
